core_planning/twist_gate/src/twist_gate.cpp

```diff
@@ -28,66 +28,67 @@
-  auto_cmd_sub_stdmap_["ctrl_cmd"] = nh_.subscribe("/ctrl_cmd", 1, &TwistGate::ctrlCmdCallback, this);
-  auto_cmd_sub_stdmap_["lamp_cmd"] = nh_.subscribe("/lamp_cmd", 1, &TwistGate::lampCmdCallback, this);
-  auto_cmd_sub_stdmap_["state"] = nh_.subscribe("/decision_maker/state", 1, &TwistGate::stateCallback, this);
-  auto_cmd_sub_stdmap_["emergency_velocity"] =
-      nh_.subscribe("emergency_velocity", 1, &TwistGate::emergencyCmdCallback, this);
-
-  output_msg_.header.seq = 0;
-  emergency_stop_msg_.data = false;
-  health_checker_ptr_->ENABLE();
-  health_checker_ptr_->NODE_ACTIVATE();
-
-  emergency_handling_time_ = ros::Time::now();
+
+    //Declare parameters
+    use_decision_maker_ = declare_parameter<bool>("~use_decision_maker", use_decision_maker_);
+    use_lgsim_ = declare_parameter<bool>("use_lgsim", use_lgsim_);
+    use_twist_ = declare_parameter<bool>("use_twist", use_twist_);
+
 }
 
-void TwistGate::twistCmdCallback(const geometry_msgs::TwistStamped::ConstPtr& input_msg)
+carma_ros2_utils::CallbackReturn TwistGate::handle_on_configure(const rclcpp_lifecycle::State &prev_state)
 {
+    //get parameters
+    get_parameter<bool>("use_decision_maker", use_decision_maker_);
+    get_parameter<bool>("use_lgsim", use_lgsim_);
+    get_parameter<bool>("use_twist", use_twist_);
+    
+    // TODO: control_command_pub_ is unused
+    control_command_pub_ = create_publisher<std_msgs::msg::String>("ctrl_mode", 1);
+    vehicle_cmd_pub_ = create_publisher<vehicle_cmd_msg_t>("vehicle_cmd",1);
+
+    config_sub_ = create_subscription<autoware_config_msgs::msg::ConfigTwistFilter>("config/twist_filter", 1, std::bind(&TwistGate::configCallback, this, std_ph::_1));
+    twist_cmd_sub_ = create_subscription<geometry_msgs::msg::TwistStamped>("/twist_cmd", 1, std::bind(&TwistGate::twistCmdCallback, this, std_ph::_1));
+    ctrl_cmd_sub_ = create_subscription<autoware_msgs::msg::ControlCommandStamped>("/ctrl_cmd", 1 , std::bind(&TwistGate::ctrlCmdCallback, this, std_ph::_1));
+    lamp_cmd_sub_ = create_subscription<autoware_msgs::msg::LampCmd>("/lamp_cmd", 1, std::bind(&TwistGate::lampCmdCallback, this, std_ph::_1));
+    state_sub_ = create_subscription<std_msgs::msg::String>("/decision_maker/state", 1, std::bind(&TwistGate::stateCallback, this, std_ph::_1));
+    vehicle_cmd_sub_ = create_subscription<vehicle_cmd_msg_t>("emergency_velocity", 1, std::bind(&TwistGate::emergencyCmdCallback, this, std_ph::_1));
+
+    emergency_stop_msg_.data = false;
+
+    emergency_handling_time_ = this->now();
+
+    return CallbackReturn::SUCCESS;
+
+}
+
+void TwistGate::twistCmdCallback(const geometry_msgs::msg::TwistStamped::UniquePtr input_msg)
+{  
   if (!use_twist_) {
     return; // Ignore message if not using twist as input
   }
-  health_checker_ptr_->CHECK_RATE("topic_rate_twist_cmd_slow", 8, 5, 1, "topic twist_cmd subscribe rate slow.");
-  health_checker_ptr_->CHECK_MAX_VALUE("twist_cmd_linear_high", input_msg->twist.linear.x,
```

Comment:
Is there functionality here which we are missing now?

Reply:
It looks like this was checking the topic rate and value and throwing warnings. That is missing now